exercises/micro-blog/description.md

```diff
@@ -0,0 +1,40 @@
+You have identified a gap in the social media market for very very short
+posts. Now that Twitter allows 280 character posts, people wanting quick
+social media updates aren't being served. You decide to create your own
+social media network.
+
+To make your product noteworthy, you make it extreme and only allow posts
+of 5 or less characters. Any posts of more than 5 characters should be
+truncated to 5.
+
+To allow your users to express themselves fully, you allow Emoji and
+other Unicode.
+
+The task is to truncate input strings to 5 characters.
+
+## Text Encodings
+
+Text stored digitally has to be converted to a series of bytes.
+There are 3 ways to map characters to bytes in common use.
+* **ASCII** can encode English language characters. All
+characters are precisely 1 byte long.
+* **UTF-8** is a Unicode text encoding. Characters take between 1
+and 4 bytes.
+* **UTF-16** is a Unicode text encoding. Characters are either 2 or
+4 bytes long.
+
+UTF-8 and UTF-16 are both Unicode encodings which means they're capable of
+representing a massive range of characters including:
+* English language text
+* Foreign language text
```

Comment:
I don't like that this assumes the perspective of a native English speaker. English is a foreign language to most of the world. Perhaps something along the lines of "text in most of the world's languages and scripts" would be a better description.

Reply:
But UTF-8 was designed to be best with English so that ís something that can be mentioned.

Reply:
It definitely can be mentioned that UTF-8 was designed to be compatible to ASCII but that can be phrased in a better way that doesn't have the issue above.

Reply:
I've incorporated the suggestion about not assuming native English speakers d2f65b4c29ea1710f9d8fe0e0dcb44fc948a3b90

I've not put in anything to do with UTF-8 being able to represent English in the least space or its backwards compatibility with ASCII. Whilst they're nice things to know about UTF-8, I'm not sure they'd help students solve the exercise.